server/container_create_linux.go

```diff
@@ -229,6 +231,33 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 		someRepoDigest = imgResult.RepoDigests[0]
 	}
 
+	systemCtx, err := s.contextForNamespace(sb.Metadata().Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get context for namespace: %w", err)
+	}
+
+	userSpecifiedImage := ctr.Config().GetImage().UserSpecifiedImage
+
+	if userSpecifiedImage == "" {
+		// Attempt to check if the image is a checkpoint OCI image
+		if _, err := s.checkIfCheckpointOCIImage(ctx, userRequestedImage); err != nil {
```

Comment:
Oh, and, of course: This is looking up `userRequestedImage` _again_, so it is racy against tag changes. It should be only working with `imgResult.ID`.

---

After `imgResult.ID` is first read, a
```go
// == Image lookup done.
// == NEVER USE userRequestedImage for anything but diagnostic logging past this point; it might
// resolve to a different image.
```
comment would not urt.

Reply:
#8590 adds that comment.